build/tsc.gni

```diff
@@ -0,0 +1,50 @@
+import("npm.gni")
+
+template("typescript_build") {
+  assert(defined(invoker.tsconfig), "Need tsconfig name to run")
+  assert(defined(invoker.sources), "Need tsc sources to run")
+  assert(defined(invoker.output_dir_name),
+         "Need output_dir_name to run, should be 'lib' or other top level dir")
+  assert(defined(invoker.output_gen_dir),
+         "Need output_gen_dir to run, should be relative to the root gen dir")
+
+  npm_action(target_name) {
+    forward_variables_from(invoker,
+                           [
+                             "deps",
+                             "public_deps",
+                             "outputs",
+                           ])
+    script = "tsc"
+    if (!defined(deps)) {
+      deps = []
+    }
+    deps += [ ":build_electron_definitions" ]
```

Comment:
eek this isn't great. this means that this `template` will _only work in `//electron/BUILD.gn`_.

If that's your intention, just move this template into `//electron/BUILD.gn`.